src/components/SetupTab/ArgsForm/index.jsx

```diff
@@ -8,35 +8,93 @@ import ArgInput from '../ArgInput';
 import { fetchDatastackFromFile } from '../../../server_requests';
 import { boolStringToBoolean } from '../../../utils';
 
-function dragoverHandler(event) {
+/** Prevent the default case for onDragOver so onDrop event will be fired. */
+function dragOverHandler(event) {
   event.preventDefault();
-  event.dataTransfer.dropEffect = 'move';
+  event.stopPropagation();
+  event.dataTransfer.dropEffect = 'copy';
 }
 
 /** Renders a form with a list of input components. */
 export default class ArgsForm extends React.Component {
   constructor(props) {
     super(props);
     this.handleChange = this.handleChange.bind(this);
+    this.inputDropHandler = this.inputDropHandler.bind(this);
     this.handleBoolChange = this.handleBoolChange.bind(this);
     this.selectFile = this.selectFile.bind(this);
-    this.onDragDrop = this.onDragDrop.bind(this);
+    this.onArchiveDragDrop = this.onArchiveDragDrop.bind(this);
+    this.dragEnterHandler = this.dragEnterHandler.bind(this);
+    this.dragLeaveHandler = this.dragLeaveHandler.bind(this);
+    this.formRef = React.createRef(); // For dragging CSS
+    this.dragDepth = 0;  // To determine Form dragging CSS 
   }
 
-  async onDragDrop(event) {
+  async onArchiveDragDrop(event) {
     /** Handle drag-drop of datastack JSON files and InVEST logfiles */
     event.preventDefault();
+    event.stopPropagation();
+    // No longer dragging so reset dragging depth and remove CSS
+    this.dragDepth = 0;
+    const formElement = this.formRef.current;
+    formElement.classList.remove("dragging");
 
     const fileList = event.dataTransfer.files;
     if (fileList.length !== 1) {
-      throw alert('only drop one file at a time.');
+      alert('Only drop one file at a time.');
+      return;
     }
     const datastack = await fetchDatastackFromFile(fileList[0].path);
 
     if (datastack.module_name === this.props.pyModuleName) {
       this.props.batchUpdateArgs(datastack.args);
     } else {
-      throw alert(`Parameter/Log file for ${datastack.module_name} does not match this model: ${this.props.pyModuleName}`)
+      alert(
+        `Parameter/Log file for ${datastack.module_name} does not match this model: ${this.props.pyModuleName}`);
```

Comment:
I'm still not crazy about this message, but it's fine to wait until we do more UX-related stuff.

Reply:
I know! Your prior comment made me change it, then Dave's comment made me change it back, because it is very descriptive of "how did I end up here?". But yea, we can sort some of these UX related things out in time!